src/components/CreateForm.tsx

```diff
@@ -56,39 +56,58 @@ class CreateForm extends React.Component<CreateScheduleProps, CreateScheduleStat
                         history.push('/vote?key=' + key + '&b=' + r.receipt.blockNumber)
                     })
             })
+            
+            // `.then()` w js działa zarówno jak map, jak i flatMap, zatem mozna to zrobic nieco ładniej (specjalnie zakomentowane, zeby latwiej bylo Ci sie polapac, co dodałem):
+            // this.props.contract
+            //     .deployed()
+            //     .then(instance => instance.create(ballotId, this.state.name, this.state.options, { from: this.props.account, gas: 1334400 }))
+            //     .then(r => {
+            //         console.log("Contract executed")
+            //         history.push('/vote?key=' + key + '&b=' + r.receipt.blockNumber)
+            //     });
         }
     }
 
     addNewDate() {
         if (this.validInput()) {
+            // ŹLE! w linii 76 modyfikujesz stan bezposrednio, a tak nie wolno!
+            // przeciez newOptions wskazuje na this.state.options :)
             let newOptions: Array<string> = this.state.options
```

Comment:
Dzięki, faktycznie wtopa. Jestem przyzwyczajony, że jak dodaję coś do kolekcji to wynikiem jest nowa kolekcja, a stara pozostaje nieruszona :)